graph/secret.go

```diff
@@ -1,10 +1,76 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
 package graph
 
-// Secret defines a wrapper to translate Azure Key Vault secrets to environment variables.
+import (
+	"github.com/Azure/acr-builder/util"
+	"github.com/pkg/errors"
+)
+
+var (
+	errMissingSecretID       = errors.New("secret is missing an ID")
+	errMissingSecretProps    = errors.New("secret is missing required akv property")
+	errSecretIDContainsSpace = errors.New("secret ID cannot contain spaces")
+	errInvalidUUID           = errors.New("The msi client ID is not a valid guid")
+)
+
+// SecretValue represents the resolved value of the Secret.
+type SecretValue struct {
+	ID    string
+	Value string
+}
+
+// Secret defines a wrapper to resolve vault secrets to values.
 type Secret struct {
-	Akv       string `yaml:"akv,omitempty"`
-	SecretEnv string `yaml:"secretEnv,omitempty"`
+	ID          string `yaml:"id"`
+	Akv         string `yaml:"akv,omitempty"`
+	MsiClientID string `yaml:"client-id,omitempty"`
+
```

Comment:
I think we avoid using `-` in Task. https://docs.microsoft.com/en-us/azure/container-registry/container-registry-tasks-reference-yaml

Reply:
clientID or msiClientID?

Reply:
`clientID` seems fine, we can leave the property as-is and refactor it if we support multiple in the future